File: sla/sla.go
package main
import (
    "fmt"
)

func main() {
    var num1, num2, resultado float64
    var op string

    fmt.Print("Digite o primeiro numero: ")
    fmt.Scan(&num1)

    fmt.Print("Digite o operador (+, -, *, /): ")
    fmt.Scan(&op)

    fmt.Print("Digite o segundo numero: ")
    fmt.Scan(&num2)

    if op == "+" {
        resultado = num1 + num2
    } else if op == "-" {
        resultado = num1 - num2
    } else if op == "*" {
        resultado = num1 * num2
    } else if op == "/" {
        if num2 != 0 {
            resultado = num1 / num2
        } else {
            fmt.Println("Erro: divisao por zero!")
            return
        }
    } else {
        fmt.Println("Operador invalido!")
        return
    }

    fmt.Println("Resultado:", resultado)
}

File: sla/sla.rs
use std::io;

fn main() {
    let mut input = String::new();

    println!("Digite o primeiro numero:");
    io::stdin().read_line(&mut input).unwrap();
    let num1: f64 = input.trim().parse().unwrap();

    input.clear();
    println!("Digite o operador (+, -, *, /):");
    io::stdin().read_line(&mut input).unwrap();
    let op = input.trim().chars().next().unwrap();

    input.clear();
    println!("Digite o segundo numero:");
    io::stdin().read_line(&mut input).unwrap();
    let num2: f64 = input.trim().parse().unwrap();

    let resultado = match op {
        '+' => num1 + num2,
        '-' => num1 - num2,
        '*' => num1 * num2,
        '/' => {
            if num2 != 0.0 { num1 / num2 }
            else { println!("Erro: divisao por zero!"); return; }
        }
        _ => { println!("Operador invalido!"); return; }
    };

    println!("Resultado: {}", resultado);
}
